frontend/src/api/appwriteApi.js

```diff
@@ -9,16 +9,10 @@ import { AppwriteServer, domainName, adminTeamName } from "../utils/config";
  * Implements convenient wrappers to manage the communication with appwrite.
  */
 let api = {
-	sdk: null,
+	sdk: new Appwrite().setEndpoint(AppwriteServer.endpoint).setProject(AppwriteServer.project),
 
-	provider: () => {
-		if (api.sdk) {
-			return api.sdk;
-		}
-		let appwrite = new Appwrite();
-		appwrite.setEndpoint(AppwriteServer.endpoint).setProject(AppwriteServer.project);
-		api.sdk = appwrite;
-		return appwrite;
+	provider() {
+		return this.sdk;
```

Comment:
Why did you do this? The previous code was based on the official React example of AppWrite: https://github.com/appwrite/demo-todo-with-react/blob/main/src/api/api.js. By default, sdk stays null and no computation happens at all until provider() was called. The way you changed it sdk is always initialized, even if not needed at all. I think we should keep it the old way.

Reply:
Is there a particular reason for not being initialized at the start?
It makes code more difficult to understand and the if-block is less efficient in the long-term.

Functionally, I didn't notice a difference. I found it more reasonable to directly initialize it at the start.

Well yes, the official Appwrite Code that I looked at doesn't look like good example code in general.